### ISSUE TITLE
Tiledarray does not build with "--elemental" and automatic madness build

### ISSUE THREAD

**Issue by shiozaki** [ValeevGroup/tiledarray#48]
```
../configure --elemental=/home/shiozaki/local
```
to get
```
-- Pulling MADNESS from: https://github.com/m-a-d-n-e-s-s/madness.git
Cloning into 'madness'...
remote: Counting objects: 37995, done.
remote: Total 37995 (delta 0), reused 0 (delta 0), pack-reused 37995
Receiving objects: 100% (37995/37995), 66.69 MiB | 31.57 MiB/s, done.
Resolving deltas: 100% (29900/29900), done.
Checking connectivity... done.
Note: checking out 'a62db8ac775573ced45095736cda85c8fda36935'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at a62db8a... Fixed a few bugs in Stack. Added unit tests.
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, `config'.
libtoolize: linking file `config/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIR, `config'.
libtoolize: linking file `config/libtool.m4'
libtoolize: linking file `config/ltoptions.m4'
libtoolize: linking file `config/ltsugar.m4'
libtoolize: linking file `config/ltversion.m4'
libtoolize: linking file `config/lt~obsolete.m4'
configure.ac:46: installing 'config/compile'
configure.ac:37: installing 'config/config.guess'
configure.ac:37: installing 'config/config.sub'
configure.ac:14: installing 'config/install-sh'
configure.ac:14: installing 'config/missing'
src/apps/chem/Makefile.am: installing 'config/depcomp'
parallel-tests: installing 'config/test-driver'
-- Looking for madness.h
-- Looking for madness.h - found
-- Will pull Elemental from https://github.com/elemental/Elemental.git
configure: WARNING: you should use --build, --host, --target
configure: WARNING: you should use --build, --host, --target
configure: WARNING: you should use --build, --host, --target
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
configure: HOST INFORMATION:   x86_64-unknown-linux-gnu    x86_64    unknown    linux-gnu
checking for style of include used by make... GNU
checking for gcc... /usr/local/gcc/gcc-5.2.0/bin/gcc
checking whether the C compiler works... no
configure: error: in `/home/shiozaki/packages/tiledarray/obj/external/build/madness':
configure: error: C compiler cannot create executables
See `config.log' for more details
CMake Error at external/madness.cmake:360 (message):
  The MADNESS configure script failed.
Call Stack (most recent call first):
  CMakeLists.txt:163 (include)
```
According to config.log, Madness' configure is trying to do
```
  $ /home/shiozaki/packages/tiledarray/external/src/madness/configure --prefix=/home/shiozaki/packages/tiledarray/obj --enable-shared --disable-static --disable-debugging --disable-optimization --disable-optimal --disable-warning --enable-madex=throw --enable-dependency-tracking --with-mpi-thread=multiple --with-fortran-int32=yes --with-stubmpi=no --with-elemental=yes --with-tbb=yes   --without-mkl --without-libxc --without-google-perf  MPICXX=/usr/local/gcc/gcc-5.2.0/bin/g++ MPICC=/usr/local/gcc/gcc-5.2.0/bin/gcc CPPFLAGS=-I/home/shiozaki/packages/tiledarray/obj/external/build/elemental/include -I/opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mpi/intel64/include CC=/usr/local/gcc/gcc-5.2.0/bin/gcc CFLAGS=-O3 -DNDEBUG CXX=/usr/local/gcc/gcc-5.2.0/bin/g++ CXXFLAGS=-std=c++11 -O3 -DNDEBUG F77= FFLAGS= LDFLAGS= LIBS=/home/shiozaki/packages/tiledarray/obj/external/build/elemental/libelemental.a /home/shiozaki/packages/tiledarray/obj/external/build/elemental/external/pmrrr/libpmrrr.a /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mpi/intel64/lib/libmpifort.so /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mpi/intel64/lib/release_mt/libmpi.so /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mpi/intel64/lib/libmpigi.a /usr/lib64/libdl.so /usr/lib64/librt.so /usr/lib64/libpthread.so /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mkl/lib/intel64/libmkl_gf_lp64.so /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mkl/lib/intel64/libmkl_intel_lp64.so /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mkl/lib/intel64/libmkl_gnu_thread.so /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mkl/lib/intel64/libmkl_core.so /usr/local/lib64/libgomp.so -lpthread -lm /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mkl/lib/intel64/libmkl_intel_lp64.so /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mkl/lib/intel64/libmkl_gnu_thread.so /opt/intel/mkl-11.3/compilers_and_libraries_2016.0.109/linux/mkl/lib/intel64/libmkl_core.so /usr/local/lib64/libgomp.so -lpthread
```
the important part is:
```
CPPFLAGS=-I/home/shiozaki/packages/tiledarray/obj/external/build/elemental/include
LIBS=/home/shiozaki/packages/tiledarray/obj/external/build/elemental/libelemental.a
```
which is not what I intended. Namely I suspect that the path is not properly passed to madness' configure script. These directories are non-existent.

**Comment by shiozaki** [comment]
I would be more than happy if tiledarray allows madness to download Elemental automatically (which seems possible)

**Comment by poulson** [comment]
Let me know if I can help with this issue in some way.

**Comment by justusc** [comment]
The configure script was written to download and build Elemental, but this option is currently broken, hence the error. Unfortunately, it is not going to be a quick fix.

As a workaround, you can build Elemental and Madness (with Elemental support) outside TA. Note that MADNESS only works with Elemental v0.84.

I recently added a CMake configure script to MADNESS that builds Elemental. I think it is working. However, it is a work in progress, and has not been thoroughly tested.

**Comment by evaleev** [comment]
This issue is entangled with [MADNESS numerical libraries requiring a particular version of Elemental](https://github.com/m-a-d-n-e-s-s/madness/issues/166). TA only uses MADworld, so in principle we can work around this. Ultimately requires switch to MADNESS cmake build.

**Comment by evaleev** [comment]
almost fully resolved in 3f3f06f718bd0f96490e694cb92786cda067c46c ... will close after pull-request and merge
